src/vm/functions/mod.rs

```diff
@@ -206,13 +205,13 @@ fn native_hash160(args: &[Value]) -> Result<Value> {
     use util::hash::Hash160;
 
     if !(args.len() == 1) {
-        return Err(Error::new(ErrType::InvalidArguments("Wrong number of arguments to hash160 (expects 1)".to_string())))
+        return Err(UncheckedError::InvalidArguments("Wrong number of arguments to hash160 (expects 1)".to_string()).into())
     }
     let input = &args[0];
     let bytes = match input {
         Value::Int(value) => Ok(value.to_le_bytes().to_vec()),
         Value::Buffer(value) => Ok(value.data.clone()),
-        _ => Err(Error::new(ErrType::NotImplemented))
+        _ => Err(RuntimeErrorType::NotImplemented)
```

Comment:
To be clear, when this ships in prod, there will be no `NotImplemented` errors in the codebase?  Asking because if the only supported types here will ever be `Int` and `Buffer`, we might as well start calling the occurrence of anything else an invalid argument.

Reply:
Yeah -- good point. This is more appropriately a TypeError. Will update.

Reply:
Changed to a TypeError, and enforced in the type checker as well.